File: Graph/617. Merge Two Binary Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def mergeTrees(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root1 or not root2:
            return root1 or root2
        from queue import Queue
        queue1 = Queue()
        queue2 = Queue()
        queue1.put(root1)
        queue2.put(root2)
        while not queue1.empty() and not queue2.empty():
            node1, node2 = queue1.get(), queue2.get()
            if node1 and node2:
                node1.val += node2.val
                if node1.left == None and node2.left != None:
                    node1.left = TreeNode(0)
                if node1.right == None and node2.right != None:
                    node1.right = TreeNode(0)
                queue1.put(node1.left)
                queue1.put(node1.right)
                queue2.put(node2.left)
                queue2.put(node2.right)
        return root1

File: Graph/617. Merge Two Binary Trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* root1, TreeNode* root2) {
        if (root1 == NULL)
            return root2;
        if (root2 == NULL)
            return root1;
        queue<TreeNode*> q1, q2;
        q1.push(root1);
        q2.push(root2);
        while (not q1.empty() && not q2.empty()){
            TreeNode* n1 = q1.front();
            TreeNode* n2 = q2.front();
            q1.pop();
            q2.pop();
            if (n1 != NULL && n2 != NULL){
                n1->val += n2->val;
                if (n1->left == NULL && n2->left != NULL)
                    n1->left = new TreeNode(0);
                if (n1->right == NULL && n2->right != NULL)
                    n1->right = new TreeNode(0);
                q1.push(n1->left);
                q1.push(n1->right);
                q2.push(n2->left);
                q2.push(n2->right);
            }
        }
        return root1;
    }
};